File: Templates/SpiderEye.Templates/templates/MyApp.VBNet/MyApp.Core/ProgramBase.vb
Imports SpiderEye

Public MustInherit Class ProgramBase
    Protected Shared Sub Run()
        Using window = New Window()
            window.Title = "MyApp"
            window.Icon = AppIcon.FromFile("icon", ".")

            Application.ContentProvider = New EmbeddedContentProvider("App")
            Application.Run(window, "index.html")
        End Using
    End Sub
End Class

File: Templates/SpiderEye.Templates/templates/MyApp.CSharp/MyApp.Core/ProgramBase.cs
using SpiderEye;

namespace MyApp.Core
{
    public abstract class ProgramBase
    {
        protected static void Run()
        {
            using (var window = new Window())
            {
                window.Title = "MyApp";
                window.Icon = AppIcon.FromFile("icon", ".");

                Application.ContentProvider = new EmbeddedContentProvider("App");
                Application.Run(window, "index.html");
            }
        }
    }
}
